supermarq-benchmarks/supermarq/run_supermarq.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime
+from typing import Dict, List, Tuple
+
+import cirq
+import cirq_superstaq as css
+
+import supermarq
+
+BENCHMARKS: List[Tuple[supermarq.benchmark.Benchmark, str]] = [
+    (supermarq.ghz.GHZ(5), "ghz5"),
+    (supermarq.hamiltonian_simulation.HamiltonianSimulation(4), "hsim4"),
+    (supermarq.mermin_bell.MerminBell(3), "mb3"),
+    (supermarq.bit_code.BitCode(3, 3, [1, 0, 1]), "bitcode3"),
+]
+
+
+def get_qpu_targets(targets: Dict[str, List[str]]) -> List[str]:
+    """Get real device targets.
+
+    Args:
+        targets: Output from `service.get_targets()`.
+
+    Returns:
+        A list with the targets corresponding to real devices.
+    """
+    qpu_targets: List[str] = []
+    run_targets = targets.get("compile-and-run")
+
+    if run_targets is None:
+        return qpu_targets
+
+    for t in run_targets:
+        if t.endswith("qpu"):
+            qpu_targets.append(t)
+    return qpu_targets
+
+
+service = css.Service()
+targets = service.get_targets()
+qpu_targets = get_qpu_targets(targets)
+
+for target in qpu_targets:
+    for benchmark, label in BENCHMARKS:
+        date = datetime.today().strftime("%Y-%m-%d")
+        tag = f"{date}-{label}-{target}"
+        print(tag)
+
+        # Since there is no way to determine if a target supports mid-circuit measurement purely
+        # through `service.target_info()`, we use a `try` block to catch when we submit an invalid
+        # circuit to a device that does not support it.
+        # Related issue: https://github.com/Infleqtion/client-superstaq/issues/550.
+
+        try:
+            circuit = benchmark.circuit()
+            if isinstance(circuit, cirq.Circuit):
+                job = service.create_job(
+                    circuit,
+                    repetitions=1000,
+                    target=target,
+                    options={"tag": tag, "qiskit_pulse": False, "lifespan": 3653},
```

Comment:
options should just be kwargs now:
```suggestion
                    tag=tag,
                    qiskit_pulse=False,
                    lifespan=3653,
```
also out of curiosity why aren't we use qiskit pulse for these?

Reply:
>  why aren't we use qiskit pulse for these?

Just cause the first Supermarq run didn't. Happy to revise.

Reply:
> also out of curiosity why aren't we use qiskit pulse for these?

some of the benchmarking circuits contain resets (product of mid-circuit measurements) that are not supported by pulse (https://github.com/Infleqtion/server-superstaq/issues/2344)

Reply:
imo whenever possible we should use the default configuration of each target (i.e. what users would get with `qiskit_pulse=None`, which is uses a pulse device whenever possible because that's where our strongest optimizations are). would it make sense to check if the circuit has resets, and just set `qiskit_pulse=False` if it does?

it also might make sense to have the server do this automatically, i.e. have it automatically use the non-pulse device if qiskit_pulse is None and the circuit contains resets

Reply:
> it also might make sense to have the server do this automatically, i.e. have it automatically use the non-pulse device if qiskit_pulse is None and the circuit contains resets

Yeah, I also think this makes sense (opened an issue for this https://github.com/Infleqtion/server-superstaq/issues/2414). Once that gets done we can remove the `qiskit_pulse` arg here.